apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -16,90 +11,89 @@
  * Name of the service for mozMobileMessage API shim.
  * @type {string}
  */
-const SERVICE_NAME = 'mozMobileMessageShim';
+const SERVICE_NAME = 'moz-mobile-message-shim';
 
 /**
- * Array of the event names that is corresponding to mozMobileMessage events.
- * @type {Array.<string>}
+ * Event set that key is corresponding to mozMobileMessage events and value is
+ * the name that broadcasts to client side.
+ * @type {Object.<string, string>}
  */
-const EVENTS = ['received', 'sending', 'sent', 'failed', 'deleted',
-                'readsuccess', 'deliverysuccess'];
+const EVENTS = Object.freeze({
+  'received': 'message-received',
+  'sending': 'message-sending',
+  'sent': 'message-sent',
+  'failed': 'message-failed-to-send',
+  'deleted': 'threads-deleted',
+  'readsuccess': 'message-read',
+  'deliverysuccess': 'message-delivered'
+});
 
 /**
  * Array of method names that need to be exposed for API shim.
  * @type {Array.<string>}
  */
-const METHODS = ['getMessage', 'retrieveMMS', 'send', 'sendMMS',
-                 'delete', 'markMessageRead', 'getSegmentInfoForText'];
+const METHODS = Object.freeze(['getMessage', 'retrieveMMS', 'send', 'sendMMS',
+  'delete', 'markMessageRead', 'getSegmentInfoForText']);
 
 /**
  * Array of stream names that need to return data (messages/threads) in chunk.
  * @type {Array.<string>}
  */
-const STREAMS = ['getThreads', 'getMessages'];
+const STREAMS = Object.freeze(['getThreads', 'getMessages']);
 
 var mozMobileMessage = null;
 
-var service = null;
-
 var MozMobileMessageShim = {
   init(mobileMessage) {
     if (!mobileMessage) {
       return;
     }
 
-    mozMobileMessage = mobileMessage;
-    service = bridge.service(SERVICE_NAME);
+    function capitalize(str) { 
+      return str[0].toUpperCase() + str.slice(1);
+    }
 
-    service.plugin(streamService);
+    var endPoint = new BroadcastChannel(SERVICE_NAME + '-channel');
+    mozMobileMessage = mobileMessage;
+    this.initService(endPoint);
 
-    EVENTS.forEach((event) => {
+    Object.keys(EVENTS).forEach((event) => {
       mozMobileMessage.addEventListener(
         event,
-        this[Utils.camelCase(`on-${event}`)]
+        this['on' + capitalize(event)].bind(this)
```

Comment:
Ah, it's for remove the dependency for Utils since it's possible that we don't need any utils for shim host(and yes I forgot to remove the Utils loading in shim host...)